packages/astro/src/@types/astro.ts

```diff
@@ -1922,6 +1935,46 @@ export interface AstroUserConfig {
 				origin?: boolean;
 			};
 		};
+
+		/**
+		 * @docs
+		 * @name experimental.rewriting
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 4.8.0
+		 * @description
+		 *
+		 * Enables the use of rewriting features in Astro pages, Endpoints and Astro middleware:
+		 *
```

Comment:
Before jumping into the code examples, can we have a code block showing enabling this in `astro.config.mjs`?

(Pls double check this formatting!)

```suggestion
		 *
		 * ```js
		 * {
		 *   experimental: {
		 *     rewriting: true,
		 *   },
		 * }
		 * ```
		 *
```